internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -447,13 +459,38 @@ func (r *Reconciler) MachineSetToDeployments(ctx context.Context, o client.Objec
 	return result
 }
 
+// getTemplatesAndSetOwner reconciles the templates referenced by a MachineDeployment ensuring they are owned by the Cluster.
+func (r *Reconciler) getTemplatesAndSetOwner(ctx context.Context, s *scope) error {
+	md := s.machineDeployment
+	cluster := s.cluster
+
+	// Make sure to reconcile the external infrastructure reference.
+	if err := reconcileExternalTemplateReference(ctx, r.Client, cluster, &md.Spec.Template.Spec.InfrastructureRef); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		s.infrastructureTemplateNotFound = true
```

Comment:
IIRC we wanted to block creations of new MachineSets (that are never going to work with missing templates) when one of the templates was not found. I think this is not implemented